src/cunumeric/set/zip_indices.h

```diff
@@ -0,0 +1,32 @@
+namespace cunumeric {
+
+using namespace legate;
+
+template <typename VAL>
+struct ZippedIndex {
+  VAL value;
+  int64_t index;
+};
+
+// Surprisingly it seems as though thrust can't figure out this comparison
```

Comment:
You might not need this if you either:
1) implement comparison overloads on `ZippedIndex`
2) use a builtin type like `std::pair` instead, which has equality implemented already